library/src/lib/calendar/calendar-views/calendar-year-view/calendar-year-view.component.ts

```diff
@@ -92,14 +100,19 @@ export class CalendarYearViewComponent implements AfterViewChecked, OnInit, OnDe
         this.onDestroy$.complete();
     }
 
-    /** @hidden */
-    constructor(private eRef: ElementRef, private calendarService: CalendarService) { }
-
-    /** @hidden */
-    private constructYearList(): void {
-        this.calendarYearList = [];
-        for (let x = 0; x < 12; x++) {
-            this.calendarYearList.push(this.firstYearInList + x);
+    /**
+     * Method that returns active cell, which means:
+     * if there is any selected year, return selected year
+     * if there is no selected year, but there is current year, return current year
+     * if there is no current year, or selected, return first one
+     */
+    private getActiveYear(): number {
+        if (this.calendarYearList.find(year => year === this.yearSelected)) {
+            return this.calendarYearList.find(year => year === this.yearSelected);
+        } else if (this.calendarYearList.find(year => year === this.currentYear)) {
+            return this.calendarYearList.find(year => year === this.currentYear);
+        } else {
+            return this.calendarYearList[0];
         }
```

Comment:
Consider instead defining these as const values and then doing your if statements on those. Not very practical nor readable to call .find twice per if